lib/vdsm/storage/exception.py

```diff
@@ -1345,25 +1361,18 @@ class VolumeGroupAlreadyExistsError(StorageException):
     msg = "Volume Group Already Exists"
 
 
-class VolumeGroupDoesNotExist(StorageException):
+class VolumeGroupDoesNotExist(VolumeLVMCommandError):
     code = 506
     msg = "Volume Group does not exist"
 
     def __init__(self, vg_name=None, vg_uuid=None, error=None):
         if vg_name is None and vg_uuid is None:
             raise ValueError("Require a VG name or UUID")
 
-        if error is not None and not isinstance(error, LVMCommandError):
-            raise TypeError(
-                f"Expecting instance of LVMCommandError, got {type(error)}")
+        super().__init__(error)
```

Comment:
So here error will be logged first, but previously we logged it last.

Reply:
I have fixed it with the value property, printing error always at the end.

It was a side effect I didn't see.